### PR TITLE
BI-1922 (Add ability to download all germplasm in a program)

### PR DESCRIPTION
**Pull request by davedrp**
# Description
[BI-1922](https://breedinginsight.atlassian.net/browse/BI-1922) Add ability to download all germplasm in a program

# Dependencies
bi-web: feature/BI-1922

# Testing
see test in [bi-web PR](https://github.com/Breeding-Insight/bi-web/pull/350)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [x] I have tested that my code works with both the brapi-java-server and BreedBase
- [ ] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: _\<please include a link to TAF run>_

[BI-1922]: https://breedinginsight.atlassian.net/browse/BI-1922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mlm483 — APPROVED** [review]

**Comment by mlm483** [comment]
@timparsons or anyone testing, I recommend changing your local docker-compose.yml to not use the `develop` tagged `brapi-server` image. Until the 2 open PRs for BI-1945 are merged ([bi-api](https://github.com/Breeding-Insight/bi-api/pull/312), [brapi-server](https://github.com/Breeding-Insight/brapi-Java-TestServer/pull/20)), the `develop` image won't work with bi-api.

For example, use the `latest` tag:
```yaml
  brapi-server:
    image: breedinginsight/brapi-java-server:latest
```

File: src/main/java/org/breedinginsight/brapi/v2/GermplasmController.java
Comment (timparsons) [inline src/main/java/org/breedinginsight/brapi/v2/GermplasmController.java:174]: This isn't needed as the logger will print the stack trace

File: src/main/java/org/breedinginsight/brapi/v2/GermplasmController.java
Comment (timparsons) [inline src/main/java/org/breedinginsight/brapi/v2/GermplasmController.java:173]: Can this be changed to `log.error` please?

**Review by timparsons — APPROVED** [review]
Approving assuming the couple of small changes are made

**Comment by davedrp** [comment]
The last commit failed unit testing.   Merged anyway.   
- here is a known bug with the Unit test.
- All previous commits passed.

**Pull request merged**
